### DISCUSSION TITLE
API rate limit exceeded for x.x.x.x.

### DISCUSSION THREAD

**Discussion by ImaJJ** [community/community#59458] [Apps, API and Webhooks] [answered]
How to solve the API rate limit exceeded for x.x.x.x. problem?

I'm new to everything about using APIs but I've got a problem. I'm using a facebook scrapper and everytime I run my code that does many requests, it shows me this error:

`{'message': "API rate limit exceeded for x.x.x.x. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)", 'documentation_url': '[https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting'}](https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting'%7D)`

I do not know where to put the Authentication and I've tried using a line like this one: 
`curl -l -H "Authorization: token XXXXXXXX" https://api.github.com/users/user`
to try to increase the API rate limit and it does show the limit increased, but when I run my code again it sends me again the error message of above.

 I'm trying to authenticate it inside my code but I don't know how.

**Comment by byrneh** [comment]
If you hit a rate limit, you should stop making requests until after the time specified by the x-ratelimit-reset header. Failure to do so may result in a lobger ban for your user.
[#dealing-with-rate-limits](https://docs.github.com/en/rest/guides/best-practices-for-integrators?apiVersion=2022-11-28#dealing-with-rate-limits)

**Comment by tshenolo** [comment, marked as answer]
Hi there

According to github documentation located at https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28
For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the person making requests.

If you want to use a token then you can use it as below:
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/user
  
  This is based on the documentation at https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28
  
  If you are coding in a specific programming language you can use tools such as Insomnia to generate code for a specific programming language. https://insomnia.rest/products/insomnia.
  

![image](https://github.com/community/community/assets/824557/0dd8dbf8-45c5-42c7-8242-23cc66a89d24)

**Comment by AkshayPratapSingh09** [comment]
Can anyone please tell me what I am Doing Wrong Here

`app.get('/api/data', async (req, res) => {
    try {
        const searchTerm = req.query.q ;
    
        let config = {
          method: 'get',
          maxBodyLength: Infinity,
          url: `https://api.github.com/search/code?q=${searchTerm}`,
         headers: { 
        'authority': 'api.github.com', 
        'accept': 'application/vnd.github+json', 
        'accept-language': 'en-US,en;q=0.9', 
        'Authorization': 'Bearer {token}', 
        // 'origin': 'https://styler-backend.vercel.app/', 
        // 'referer': 'https://styler-backend.vercel.app//', 
        'sec-ch-ua': '"Not_A Brand";v="8", "Chromium";v="120", "Google Chrome";v="120"', 
        'sec-fetch-dest': 'empty', 
        'sec-fetch-mode': 'cors', 
        'sec-fetch-site': 'cross-site', 
        'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36', 
        'x-github-api-version': '2022-11-28'
      }
    };

    // Make the Axios request
    const response = await axios(config);

    // Send the response data back to the client
    res.json(response.data);
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: 'Internal Server Error' });
  }
});
`


Unneccessary Error: Bad Request or Wrong Credentials
